product/v2.py

```diff
@@ -217,8 +217,29 @@ def write_component_descriptor_to_dir(
         raise
 
 
-def upload_component_descriptor_v2_to_oci_registry(
+def replicate_oci_artifact_and_patch_component_descriptor(
```

Comment:
not sure about the name: from the name, I would expect the CD to be patched in this function. However, the signature suggests that an already-patched component-descriptor be passed.